src/ext/java/org/opentripplanner/ext/transmodelapi/mapping/TripRequestMapper.java

```diff
@@ -0,0 +1,372 @@
+      (List<String> lines) ->
+        request
+          .journey()
+          .transit()
+          .setWhiteListedRoutes(RouteMatcher.idMatcher(mapIDsToDomain(lines)))
+    );
+    callWith.argument(
+      "banned.lines",
+      (List<String> lines) ->
+        request.journey().transit().setBannedRoutes(RouteMatcher.idMatcher(mapIDsToDomain(lines)))
+    );
+    callWith.argument(
+      "banned.serviceJourneys",
+      (Collection<String> serviceJourneys) ->
+        request.journey().transit().setBannedTrips(mapIDsToDomain(serviceJourneys))
+    );
+
+    // callWith.argument("banned.quays", quays -> request.setBannedStops(mappingUtil.prepareListOfFeedScopedId((List<String>) quays)));
+    // callWith.argument("banned.quaysHard", quaysHard -> request.setBannedStopsHard(mappingUtil.prepareListOfFeedScopedId((List<String>) quaysHard)));
+
+    callWith.argument(
+      "whiteListed.rentalNetworks",
+      (List<String> networks) -> request.journey().rental().setAllowedNetworks(Set.copyOf(networks))
+    );
+
+    callWith.argument(
+      "banned.rentalNetworks",
+      (List<String> networks) -> request.journey().rental().setBannedNetworks(Set.copyOf(networks))
+    );
+
+    // callWith.argument("heuristicStepsPerMainStep", (Integer v) -> request.heuristicStepsPerMainStep = v);
+    // callWith.argument("compactLegsByReversedSearch", (Boolean v) -> { /* not used any more */ });
+    // callWith.argument("banFirstServiceJourneysFromReuseNo", (Integer v) -> request.banFirstTripsFromReuseNo = v);
+
+    // callWith.argument("useFlex", (Boolean v) -> request.useFlexService = v);
+    // callWith.argument("ignoreMinimumBookingPeriod", (Boolean v) -> request.ignoreDrtAdvanceBookMin = v);
+
+    if (GqlUtil.hasArgument(environment, "modes")) {
+      request.journey().setModes(mapRequestModes(environment.getArgument("modes")));
+    }
+
+    request.withPreferences(preferences -> {
+      mapPreferences(environment, callWith, preferences);
+    });
+
+    /*
+        List<Map<String, ?>> transportSubmodeFilters = environment.getArgument("transportSubmodes");
+        if (transportSubmodeFilters != null) {
+            request.transportSubmodes = new HashMap<>();
+            for (Map<String, ?> transportSubmodeFilter : transportSubmodeFilters) {
+                TraverseMode transportMode = (TraverseMode) transportSubmodeFilter.get("transportMode");
+                List<TransmodelTransportSubmode> transportSubmodes = (List<TransmodelTransportSubmode>) transportSubmodeFilter.get("transportSubmodes");
+                if (!CollectionUtils.isEmpty(transportSubmodes)) {
+                    request.transportSubmodes.put(transportMode, new HashSet<>(transportSubmodes));
+                }
+            }
+        }*/
+
+    return request;
+  }
+
+  /**
+   * Create a RouteViaRequest from the input fields of the viaTrip query arguments.
+   */
+  public static RouteViaRequest createRouteViaRequest(DataFetchingEnvironment environment) {
```

Comment:
I like that this is now extracted out of the TransmodelGraphQLPlanner. We should probably split this class into multiple classes, one for each return type (ViaLocationMapper, GenericLocationMapper, and so on). The private mapper methods can be package local (there should also be unit tests on each mapping method, but that is for later/next version of the api.

Reply:
Done